File: SOLID/Java/DIP/OrderService.java
class OrderService {
    private final PaymentProcessor paymentProcessor;
    private final InventoryManager inventoryManager;

    public OrderService(PaymentProcessor paymentProcessor,
                        InventoryManager inventoryManager) {
        this.paymentProcessor = paymentProcessor;
        this.inventoryManager = inventoryManager;
    }

    public void placeOrder(String product, double amount) {
        if (paymentProcessor.pay(amount)) {
            inventoryManager.reserve(product);
            System.out.println("Order placed");
        }
    }
}

File: SOLID/TS/DIP/OrderService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderService = void 0;
var OrderService = /** @class */ (function () {
    function OrderService(paymentProcessor, inventoryManager) {
        this.paymentProcessor = paymentProcessor;
        this.inventoryManager = inventoryManager;
    }
    OrderService.prototype.placeOrder = function (product, amount) {
        if (this.paymentProcessor.pay(amount)) {
            this.inventoryManager.reserve(product);
            console.log("Order placed");
        }
    };
    return OrderService;
}());
exports.OrderService = OrderService;
